plugins/performance-lab/includes/admin/plugin-activate-ajax.js

```diff
@@ -0,0 +1,175 @@
+					}
+
+					// Replace the 'Activate' button with a disabled 'Active' button.
+					target.replaceWith(
+						$( '<button>', {
+							type: 'button',
+							class: 'button button-disabled',
+							disabled: true,
+							text: __( 'Active', 'performance-lab' ),
+						} )
+					);
+
+					const pluginSettingsURL =
+						responseData?.data?.pluginSettingsURL;
+
+					// Select the container for action buttons related to the plugin.
+					const actionButtonList = $(
+						`.plugin-card-${ pluginSlug } .plugin-action-buttons`
+					);
+
+					if ( pluginSettingsURL && actionButtonList ) {
+						// Append a 'Settings' link to the action buttons.
+						actionButtonList.append(
+							$( '<li>' ).append(
+								$( '<a>', {
+									href: pluginSettingsURL,
+									text: __( 'Settings', 'performance-lab' ),
+								} )
+							)
+						);
+					}
+
+					showAdminNotice(
+						__( 'Feature activated.', 'performance-lab' ),
+						'success',
+						pluginSettingsURL
+					);
+				},
+				error() {
+					showAdminNotice(
+						__(
+							'There was an error activating the plugin. Please try again.',
+							'performance-lab'
+						)
+					);
+
+					target
+						.removeClass( 'updating-message' )
+						.text( __( 'Activate', 'performance-lab' ) );
+				},
+			} );
+		}
+	);
+
+	/**
+	 * Displays an admin notice with the given message and type.
+	 *
+	 * @param {string} message             - The message to display in the notice.
+	 * @param {string} [type='error']      - The type of notice ('error', 'success', etc.).
+	 * @param {string} [pluginSettingsURL] - Optional URL for the plugin settings.
+	 */
+	function showAdminNotice(
+		message,
+		type = 'error',
+		pluginSettingsURL = undefined
+	) {
+		a11y.speak( message );
+
+		// Create the notice container elements.
+		const notice = $( '<div>', {
+			class: `notice is-dismissible notice-${ type }`,
+		} );
+
+		const messageWrap = $( '<p>' ).text( message );
+
+		// If a plugin settings URL is provided, append a 'Review settings.' link.
+		if ( pluginSettingsURL ) {
+			messageWrap
+				.append( ` ${ __( 'Review', 'performance-lab' ) } ` )
+				.append(
+					$( '<a>', {
+						href: pluginSettingsURL,
+						text: __( 'settings', 'performance-lab' ),
+					} )
+				)
+				.append( _x( '.', 'Punctuation mark', 'performance-lab' ) );
+		}
+
+		const dismissButton = $( '<button>', {
+			type: 'button',
+			class: 'notice-dismiss',
+			click: () => notice.remove(),
+		} ).append(
+			$( '<span>', {
+				class: 'screen-reader-text',
+				text: __( 'Dismiss this notice.', 'performance-lab' ),
+			} )
+		);
+
+		notice.append( messageWrap, dismissButton );
+
+		const noticeContainer = $( '.wrap.plugin-install-php' );
+
+		if ( noticeContainer.length ) {
+			// If the container exists, insert the notice after the first child.
+			noticeContainer.children().eq( 0 ).after( notice );
+		} else {
+			$( 'body' ).prepend( notice );
+		}
+	}
+
+	// @ts-ignore
+} )( window.jQuery );
```

Comment:
Does it make sense to make jQuery a dependency? Would plain JS not be feasible?

Reply:
+1, it would be great to use vanilla JavaScript rather than relying on jQuery, to set a good example for performance.